src/rust/engine/src/context.rs

```diff
@@ -183,9 +183,12 @@ impl Core {
           local_execution_root_dir.to_path_buf(),
           store.clone(),
           executor.clone(),
-          // TODO: The nailgun pool size should almost certainly be configurable independent
-          // of concurrency, along with per-instance memory usage.
-          exec_strategy_opts.local_parallelism,
+          // We set the nailgun pool size to twice the number that will ever be active in order to
+          // keep warm but idle processes for task fingerprints which are not currently busy (e.g.
+          // while busy running scalac, keeping some javac processes idle).
+          // TODO: The nailgun pool size should be configurable independent of concurrency, along
```

Comment:
Ticket for this task as well?

Reply:
It's all the same ticket I think... you cannot adjust the total number of instances without accounting for memory. I'll expand the description there if need be.

Reply:
Yea, it's included on #13067: the idea as it stands is to statically compute the count of JVMs from the total memory limit and per-instance settings.